File: workplace-fe/src/app/pages/sessions/model/session.ts
import {Client} from "../../clients/model/client";

export class Session {
  id: string | undefined;
  workplaceId: string | undefined;
  name: string | undefined;
  description: string | undefined;
  createdAt: Date | undefined;
  updatedAt: Date | undefined;
  startTime: Date | undefined;
  endTime: Date | undefined;
  sessionDuration: number | undefined;
  inProgress: boolean | undefined;
  expectedEndTime: Date | undefined;
  hostFullName: string | undefined;
  sessionStatus: SessionStatus | undefined;
  url: string | undefined;
  clients: Client[] = [];
  instant: boolean = false;


}

export enum SessionStatus {
  NEW = "NEW", IN_PROGRESS = "IN_PROGRESS", FINISHED = "FINISHED", ARCHIVED = "ARCHIVED", MISSED = "MISSED"
}

File: workplace-be/src/main/java/com/emetaplus/workplace/session/model/Session.java
package com.emetaplus.workplace.session.model;


import com.emetaplus.workplace.client.model.Client;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;
import lombok.experimental.FieldNameConstants;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.ManyToMany;
import javax.persistence.Table;
import javax.persistence.Transient;
import java.time.LocalDateTime;
import java.util.Set;
import java.util.UUID;

@Entity
@Getter
@Setter
@RequiredArgsConstructor
@Table(name = "sessions")
@FieldNameConstants
public class Session {

    @Id
    @GeneratedValue
    private UUID id;

    private String name;
    private String description;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private LocalDateTime startTime;
    private LocalDateTime endTime;
    //New class or just minutes ???
    private Long sessionDuration;
    private boolean inProgress = false;
    private String url;

    @Transient
    private LocalDateTime expectedEndTime;

    @Transient
    private String hostFullName;

    @Transient
    private boolean instant;

    @Enumerated(EnumType.STRING)
    @Column(name = "session_status")
    private SessionStatus sessionStatus;

    @Column(name = "workplace_id")
    private UUID workplaceId;

    @ManyToMany
    private Set<Client> clients;


}
